syntax = "proto2";

option optimize_for = LITE_RUNTIME;

// XX_REQ 客户端请求消息
// XX_ACK 针对请求消息应答
// XX_NTF 通知消息

enum MessageID {
	MSG_BEGIN					= 0;
	
    //============================================================================================
    //                 Cient <==> GateSVR
    //============================================================================================
	// 
	MSG_LOGIN_GATE_ACK			= 100001;
	
	
	
    //============================================================================================
    //                 GameSVR <==> FightSVR
    //============================================================================================
	MSG_NEW_FIGHT_REQ			= 200001;	// 新建战斗请求
	MSG_NEW_FIGHT_ACK			= 200002;
	
	
    //============================================================================================
    //                 Cient <==> FightSVR
    //============================================================================================
	MSG_JOIN_FIGHT_REQ			= 300001;	// 客户端加入游戏请求
	MSG_JOIN_FIGHT_ACK			= 300002;
	
	MSG_RUN_NEXT_FRAME_REQ		= 300003;	// 进入下一逻辑帧
	MSG_RUN_NEXT_FRAME_ACK		= 300004;
	
	MSG_PLAYER_READY_NTF		= 300005;	// 玩家准备完毕通知
	
	MSG_PLAYER_EXIT_FIGHT_NTF	= 300006;	// 玩家退出战斗通知
	
	MSG_EXIT_FIGHT_REQ			= 300007;	// 退出战斗请求(客户端正常退出逻辑)
	MSG_EXIT_FIGHT_ACK			= 300008;	// 
		
	
    //============================================================================================
    //                 Server <==> Server
    //============================================================================================
	// 
	MSG_REG_SERVER_NODE_REQ		= 600001;	// 服务器节点注册请求
	MSG_REG_SERVER_NODE_ACK		= 600002;	
	
	
}
